### PR TITLE
Don’t fail post upload when only attachment of media failed.

### LINKED ISSUE #8727: Crash on Jetpack sites with VideoPress (Atomic or other third party)
### Expected behavior

Uploading videos to an Jetpack site that has VideoPress active and then publish of the post where the video is attached should work correctly.

### Actual behavior

When uploading larges video to a site sometime after the upload completes and the publish button is pressed the publish action fails and then pressing publish again makes the app crash

### Steps to reproduce the behaviour

- Create a new post on a Jetpack site with VideoPress active
- Add a new video to it (preferably a large one, more than 100Mb size)
- Expect for the upload to finish
- The moment the upload finish, tap on Publish on the post
- See if an error message shows up
- Tap Publish again, see the app crash

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #8727 

This PR just does a small change, to the postUpload method in order to not make it fail when an attachment fails.

To test:
 - See the ticket with the step on how to reproduce the original crash.

cc @diegoreymendez i think the reason was making the app crash after, was that when post upload was successful before we attached media to it, we updating and deleting the post.revision and probably made the Post entity referred by AztecPostViewController to be delete from the MOC and made the MOC nil on the post.

### CODE REVIEW

File: WordPress/Classes/Services/PostService.m [hunk 1]
```diff
@@ -192,7 +192,12 @@ - (void)uploadPost:(AbstractPost *)post
                     if (success) {
                         success(postInContext);
                     }
-                } failure:failure];
+                } failure:^(NSError *error){
+                    // Sergio Estevao: even if media fails to attach we are answering with success because the post uploaded sucessfull and the only thing that failed what attaching the media to it.
```
Comment (elibud) [inline WordPress/Classes/Services/PostService.m]: There's a typo here, `he only thing that failed what` should be `he only thing that failed was`.

**Comment by diegoreymendez** [comment]
Wouldn't this make the post upload succeed when the media upload fails for other reasons?

It feels to me like this PR is not solving the real issue, which is that once the user taps on "remove media" the upload shouldn't try to upload media at all.

File: WordPress/Classes/Services/PostService.m [hunk 2]
```diff
@@ -192,7 +192,12 @@ - (void)uploadPost:(AbstractPost *)post
                     if (success) {
                         success(postInContext);
                     }
-                } failure:failure];
+                } failure:^(NSError *error){
+                    // Sergio Estevao (2018-02-27): even if media fails to attach we are answering with success because the post uploaded sucessfull and the only thing that failed was attaching the media to it.
+                    if (success) {
```
Comment (frosty) [inline WordPress/Classes/Services/PostService.m]: Are there not situations where we will want to report failure? Can / should we check that it's due to the attachment?

**Comment by frosty** [comment]
I've just tried multiple times and I'm not able to recreate the original issue on `develop, so I'm not sure I'm able to properly review this right now. Also, same question as Diego, will this mask other errors?

**Pull request merged**